tensorflow/core/transforms/remapper/pass.cc

```diff
@@ -84,14 +88,177 @@ class MatchMulSigmoid : public RewritePattern {
   OperationName sigmoid_name_;
 };
 
+static FailureOr<TFOp> CreateContractionWithBiasAddOp(PatternRewriter &rewriter,
+                                                      OpPropertyHelper &helper,
+                                                      Operation *contraction_op,
+                                                      Operation *bias_add_op) {
+  // Get all operands for fused op
+  Value input = contraction_op->getOperand(0);
+  Value filter = contraction_op->getOperand(1);
+  Value bias = bias_add_op->getOperand(1);
+
+  SmallVector<Value> operands;
+  operands.push_back(input);
+  operands.push_back(filter);
+  operands.push_back(bias);
+
+  // Get the contraction type, MatMul/Conv2D/DepthwiseConv2dNative etc.
+  std::string fused_contraction_type;
+  if (helper.getDialect()->IsConv2D(contraction_op)) {
+    fused_contraction_type = "tfg._FusedConv2D";
+  } else if (helper.getDialect()->IsMatMul(contraction_op)) {
+    fused_contraction_type = "tfg._FuseMatMul";
+  } else if (helper.getDialect()->IsDepthwiseConv2dNative(contraction_op)) {
+    fused_contraction_type = "tfg._FusedDepthwiseConv2dNative";
+  } else {
+    // TODO(intel-tf): Silently return. Add some loggin info that fusion is
+    // not supported.
+    return failure();
+  }
+
+  // matchAndRewrite function will set the appropriate location.
+  Operation *new_op =
+      rewriter.create(UnknownLoc::get(contraction_op->getContext()),
+                      rewriter.getStringAttr(fused_contraction_type), operands,
+                      bias_add_op->getResultTypes(), {});
+
+  // Fill in attributes
```

Comment:
There are two ways to construct an operation. In this case, you may consider to use OperationState to build the required arguments. So that you can have all the arguments(includes attrs) ready before creation.